samples/matter/common/src/clusters/identify.h

```diff
@@ -0,0 +1,184 @@
+	 *
+	 * This class is used to implement the Identify delegate for nRF Development
+	 * Kits. It is used to blink the LED2 on the nRF Development Kit.
+	 *
+	 * This class is designed to be used as a base class for custom identify
+	 * delegate implementations. Derived classes can override the virtual methods
+	 * to customize the identify behavior.
+	 *
+	 * @note By default, trigger effects are disabled. Use the constructor
+	 * parameter or SetTriggerEffectEnabled() to enable them.
+	 */
+	class IdentifyDelegateImplNrf : public chip::app::Clusters::IdentifyDelegate {
+	public:
+		/**
+		 * @brief Construct a new IdentifyDelegateImplNrf object.
+		 *
+		 * @param isTriggerEffectEnabled Whether trigger effects are enabled.
+		 *                               Defaults to false.
+		 */
+		explicit IdentifyDelegateImplNrf(bool isTriggerEffectEnabled = false,
+						 void (*customIdentifyStopCallback)(void) = nullptr)
+			: mIsTriggerEffectEnabled(isTriggerEffectEnabled),
+			  mCustomIdentifyStopCallback(customIdentifyStopCallback)
+		{
+		}
+
+		virtual void OnIdentifyStart(chip::app::Clusters::IdentifyCluster &cluster) override
+		{
+			Nrf::PostTask([] {
+				Nrf::GetBoard()
+					.GetLED(Nrf::DeviceLeds::LED2)
+					.Blink(Nrf::LedConsts::kIdentifyBlinkRate_ms);
+			});
+		}
+
+		virtual void OnIdentifyStop(chip::app::Clusters::IdentifyCluster &cluster) override
+		{
+			if (mCustomIdentifyStopCallback) {
+				mCustomIdentifyStopCallback();
+			} else {
+				Nrf::PostTask([] { Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).Set(false); });
+			}
+		}
+
+		virtual void OnTriggerEffect(chip::app::Clusters::IdentifyCluster &cluster) override
+		{
+			switch (cluster.GetEffectIdentifier()) {
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kBlink:
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kBreathe:
+				Nrf::PostTask([] {
+					Nrf::GetBoard()
+						.GetLED(Nrf::DeviceLeds::LED2)
+						.Blink(Nrf::LedConsts::kTriggerEffectStart_ms,
+						       Nrf::LedConsts::kTriggerEffectFinish_ms);
+				});
+				break;
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kOkay:
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kChannelChange:
+				Nrf::PostTask([] { Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).Set(true); });
+				break;
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kFinishEffect:
+			case chip::app::Clusters::Identify::EffectIdentifierEnum::kStopEffect:
+				if (mCustomIdentifyStopCallback) {
+					mCustomIdentifyStopCallback();
+				} else {
+					Nrf::PostTask([] { Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).Set(false); });
+				}
+				break;
+			default:
+				return;
+			}
+		}
+
+		virtual bool IsTriggerEffectEnabled() const override { return mIsTriggerEffectEnabled; }
+
+	protected:
+		bool mIsTriggerEffectEnabled = false;
+		void (*mCustomIdentifyStopCallback)(void) = nullptr;
+	};
+
+	/**
+	 * @brief Identify cluster implementation for nRF Connect SDK.
```

Comment:
Is it Identify Cluster or Identify cluster? Use the capitalization consistently.